core/src/main/scala/org/apache/spark/deploy/SparkHadoopUtil.scala

```diff
@@ -474,40 +498,79 @@ private[spark] object SparkHadoopUtil extends Logging {
 
   private def appendHiveConfigs(hadoopConf: Configuration): Unit = {
     hiveConfKeys.foreach { kv =>
-      hadoopConf.set(kv.getKey, kv.getValue)
+      hadoopConf.set(kv.getKey, kv.getValue, "Set by Spark from hive-site.xml")
     }
   }
 
   private def appendSparkHadoopConfigs(conf: SparkConf, hadoopConf: Configuration): Unit = {
     // Copy any "spark.hadoop.foo=bar" spark properties into conf as "foo=bar"
     for ((key, value) <- conf.getAll if key.startsWith("spark.hadoop.")) {
-      hadoopConf.set(key.substring("spark.hadoop.".length), value)
+      hadoopConf.set(key.substring("spark.hadoop.".length), value,
+        "Set by Spark from keys starting with 'spark.hadoop'")
     }
+    val setBySpark = "Set by Spark to default values"
```

Comment:
Could you define these constants in `object SparkHadoopUtil`, please?
- `"Set by Spark from hive-site.xml"`
- `"Set by Spark to default values"`

Reply:
happy to